### PR TITLE
Remove jQuery class from the `repo-issue.js` file

### PR DESCRIPTION
**Pull request by yardenshoham**
Switched from jQuery class functions to plain JavaScript `classList`.

Tested the following functionalities and they work as before:
- delete issue comment
- cancel code comment
- update (merge or rebase) pull request
- re-request review
- reply to code comment
- show/hide outdated comments
- add code comment
- edit issue title

### CODE REVIEW

File: web_src/js/features/repo-issue.js [hunk 1]
```diff
@@ -504,18 +518,19 @@ export function initRepoPullRequestReview() {
...
+    const path = this.closest('[data-path]')?.getAttribute('data-path');
+    const tr = this.closest('tr');
+    const lineType = tr.getAttribute('data-line-type');
 
-    let $ntr = $tr.next();
-    if (!$ntr.hasClass('add-comment')) {
+    const ntr = tr.nextElementSibling;
+    let $ntr = $(ntr);
+    if (!ntr?.classList.contains('add-comment')) {
       $ntr = $(`
```
Comment (delvh) [inline web_src/js/features/repo-issue.js:534]: Code is currently weird:
```suggestion
    if (!ntr?.classList.contains('add-comment')) {
      const $ntr = $(`
```

**Review by delvh — APPROVED** [review]

File: web_src/js/features/repo-issue.js (hunk 1 above)
Comment (delvh) [reply web_src/js/features/repo-issue.js:534]: Missed the mention in `#546`.

File: web_src/js/features/repo-issue.js [hunk 2]
```diff
@@ -158,47 +158,55 @@ export function initRepoIssueSidebarList() {
...
+        if (!parentTimelineGroup?.querySelector('.timeline-item.comment') && !parentTimelineItem?.querySelector('.conversation-holder')) {
+          const timelineAvatar = parentTimelineGroup?.querySelector('.timeline-avatar');
+          timelineAvatar?.classList.remove('timeline-avatar-offset');
         }
       } catch (error) {
         console.error(error);
       }
     }
-    return false;
+    event.preventDefault();
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]: Move it to line 163.

File: web_src/js/features/repo-issue.js [hunk 3]
```diff
@@ -222,54 +230,60 @@ export function initRepoIssueDependencyDelete() {
...
-  $(document).on('click', '.cancel-code-comment', (e) => {
-    const $form = $(e.currentTarget).closest('form');
-    if ($form.length > 0 && $form.hasClass('comment-form')) {
-      $form.addClass('tw-hidden');
-      showElem($form.closest('.comment-code-cloud').find('button.comment-form-reply'));
+  document.addEventListener('click', (e) => {
+    if (!e.target.matches('.cancel-code-comment')) return;
+
+    const form = e.target.closest('form');
+    if (form && form.classList.contains('comment-form')) {
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]:
```suggestion
    if (form?.classList.contains('comment-form')) {
```

File: web_src/js/features/repo-issue.js [hunk 4]
```diff
@@ -222,54 +230,60 @@ export function initRepoIssueDependencyDelete() {
...
-    if (data?.redirect) {
-      window.location.href = data.redirect;
-    } else if (redirect) {
-      window.location.href = redirect;
-    } else {
-      window.location.reload();
-    }
-  });
+  const pullUpdateButton = document.querySelector('.update-button > button');
+  if (pullUpdateButton) {
```
Comment (silverwind) [inline web_src/js/features/repo-issue.js]: Make this line `if (!pullUpdateButton) return;` and then dedent the code below it.

File: web_src/js/features/repo-issue.js (hunk 2 above)
Comment (yardenshoham) [reply web_src/js/features/repo-issue.js]: https://github.com/go-gitea/gitea/pull/30192/commits/a74dcbf20cdb76787ab45aa1e94ccef730555a7f

File: web_src/js/features/repo-issue.js (hunk 4 above)
Comment (yardenshoham) [reply web_src/js/features/repo-issue.js]: https://github.com/go-gitea/gitea/pull/30192/commits/e2bdb5ae2b55cb115fee12851bf2f5cbee5a2214

**Review by silverwind — APPROVED** [review]

**Pull request merged**